hs_core/management/commands/reset_file_size_system_metadata.py

```diff
@@ -0,0 +1,146 @@
+
+    def handle(self, *args, **options):
+        update = options['update'] if options['update'] else False
+        uid = options['uid'] if options['uid'] else None
+        resource_id = options['resource_id'] if options['resource_id'] else None
+        min_quota_django_model = int(options['min_quota_django_model']) if options['min_quota_django_model'] else 0
+        refreshed_weeks = options['refreshed_weeks']
+        modified_weeks = options['modified_weeks']
+
+        resources_to_modify = []
+
+        if resource_id:
+            if min_quota_django_model or uid:
+                raise CommandError("Cannot filter by resource_id in addition to min_quota_django_model or uid")
+            try:
+                res = ResourceFile.objects.get(short_id=resource_id)
+                resources_to_modify = [res]
+            except Exception as e:
+                print(f"Resource with id {resource_id} not found: {e}")
+                raise CommandError(f"Resource with id {resource_id} not found: {e}")
+        elif uid:
+            if min_quota_django_model:
+                raise CommandError("Cannot filter by uid in addition to min_quota_django_model")
+            try:
+                user = User.objects.get(id=uid, is_active=True, is_superuser=False)
+                resources_to_modify = user.uaccess.owned_resources
+            except User.DoesNotExist:
+                print(f'Active user with id {uid} not found')
+        elif min_quota_django_model > 0:
+            uqs = UserQuota.objects.filter(user__is_active=True) \
+                .filter(user__is_superuser=False)
+            num_uqs = uqs.count()
+            counter = 1
+            print(f'Found {num_uqs} users with quota above {min_quota_django_model} GB')
+            for uq in uqs:
+                if uq.used_value < min_quota_django_model:
+                    print(f"Skipping {uq.user.username}. Use={uq.used_value}GB is less than {min_quota_django_model}")
+                    continue
+                user = uq.user
+                print(f'{counter}/{num_uqs}: \
+                      Getting resources for user: {user.username}, {current_site}/user/{user.id}/')
+                owned_resources = user.uaccess.owned_resources
+                for res in owned_resources:
+                    if res.get_quota_holder() == user:
+                        resources_to_modify.append(res)
+                counter += 1
+        else:
+            res_files = ResourceFile.objects.all()
+            res_files = filter_files(res_files, refreshed_weeks=refreshed_weeks, modified_weeks=modified_weeks)
+            num_files = res_files.count()
+            print(f"Total files: {num_files}")
+            if update and num_files > 0:
+                file_counter = 1
+                for res_file in res_files.iterator():
+                    print(f"{file_counter}/{num_files}")
+                    res_file.calculate_size(resource=res_file.resource, save=True)
```

Comment:
I did it this way because the bulk update was timing out and then none of them would be saved.
Saving as we go, and updating `filesize_cache_updated` in the process is slower, but at least the incremental progress is maintained.

I guess the other approach would be to use bulk update with batches. Maybe I will test as is and if we need bulk update for speed we can do so?

Reply:
@pkdash after more testing on beta, I discovered that using `bulk_update` has almost no effect on performance of this management command. This is because the irods filesize query takes vastly longer than any .save() on the django db.

So I removed the bulk_updates. I did add the ability to "chunk" querysets, so that we can run in batches of ResourceFiles. This doesn't have any performance gain that I could detect, but it does make it a bit easier to read the log file so I decided to leave the chunking mechanism in place.

I also added some timers into the logging which was helpful for comparisons but I think is also nice when running the commands so I would like to leave those in.

So requesting your final approval, thank you for all of the reviews on this work!